### PR TITLE
gitserver: Make Cmd fields private

### PR DESCRIPTION
**Pull request by ryanslade**
This is in preparation for having two Command implementations, one that calls    
out to gitserver over the network and one that runs against a local git
directory. This will allow us to more easily test inside the gitserver package
and should remove the need to run gitserver instances in tests. It may also 
allow us to remove a lot of the mocking code we use today.
 
All fields inside of `gitserver.Cmd` were made private. Methods were introduced
where we need to read or write the values outside of the package.

In the case of the `Repo` field, we added it to the constructor as in all
places it was used previously we create a `Cmd` and then set the value of the
`Repo` field on the next line so this could be collapsed.

# Test plan

Code only moved around. All tests still pass

### CODE REVIEW

File: internal/gitserver/client.go [hunk 1]
```diff
@@ -174,7 +174,7 @@ type Client interface {
 	BatchLog(ctx context.Context, opts BatchLogOptions, callback BatchLogCallback) error
 
 	// Command creates a new Cmd. Command name must be 'git', otherwise it panics.
-	Command(name string, args ...string) *Cmd
+	Command(repo api.RepoName, name string, args ...string) *Cmd
```
Comment (vrto) [inline internal/gitserver/client.go:177]: Thinking out loud, but it appears that we always pass `"git"` for `name` parameter 🤔 Maybe we can drop the second param and call this `GitCommand` instead? (no need to tackle it in this PR)

File: internal/gitserver/client.go [hunk 2]
```diff
@@ -830,23 +830,12 @@ func repoNamesFromRepoCommits(repoCommits []api.RepoCommit) []string {
...
-	return &Cmd{
-		client: c,
-		Args:   append([]string{"git"}, arg...),
-	}
+	repo           api.RepoName // the repository to execute the command in
+	ensureRevision string
+	args           []string
+	noTimeout      bool
+	exitStatus     int
+	httpPost       func(ctx context.Context, repo api.RepoName, op string, payload interface{}) (resp *http.Response, err error)
```
Comment (vrto) [inline internal/gitserver/client.go:838]: Let's maybe give this a more descriptive name, like `execCall` or something similar?

**Review by vrto — COMMENTED** [review]
I don't 100% understand the benefit of these methods now, is that so that we can extract an interface in the next step?

**Comment by ryanslade** [comment]
> I don't 100% understand the benefit of these methods now, is that so that we can extract an interface in the next step?

Yep, the idea is that soon `client.Command` will return an interface. One implementation will be our current `Cmd` that calls out to gitserver and another that can be used in tests will run against the local disk.

**Review by vrto — APPROVED** [review]
Let's do this 👍 🎉

**Pull request merged**